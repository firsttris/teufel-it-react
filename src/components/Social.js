import React, { Component } from 'react';

class Social extends Component {
  render() {
    const linkStyle = {
      margin: '10px',
      color: '#f00',
      textDecoration: 'none'
    }
    return (
        <a style={linkStyle} className={"fa fa-2x "+this.props.icon} aria-hidden="true" target="_blank" rel="noopener noreferrer" href={this.props.url}> </a>
    );
  }
}

export default Social;
