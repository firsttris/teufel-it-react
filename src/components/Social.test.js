
import React from 'react';
import Social from './Social';
import renderer from 'react-test-renderer';

test('Display Social', () => {
  const component = renderer.create(
    <Social icon="fa-github" url="https://github.com/firsttris"></Social>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});