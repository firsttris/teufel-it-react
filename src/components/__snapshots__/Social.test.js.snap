// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Social 1`] = `
<a
  aria-hidden="true"
  className="fa fa-2x fa-github"
  href="https://github.com/firsttris"
  rel="noopener noreferrer"
  style={
    Object {
      "color": "#f00",
      "margin": "10px",
      "textDecoration": "none",
    }
  }
  target="_blank"
>
   
</a>
`;
